gravitee-apim-portal-webui-next/src/components/application-card/application-card.component.spec.ts

```diff
@@ -0,0 +1,141 @@
+    expect(titleElement.textContent).toContain(mockData.application.name);
+  });
+
+  it('should display the owner', () => {
+    const versionElement = fixture.nativeElement.querySelector('.m3-body-medium');
+    expect(versionElement.textContent).toContain(`Owner: ${mockData.application.owner?.display_name}`);
+  });
+
+  it('should display content if available', () => {
+    const contentElement = fixture.nativeElement.querySelector('.app-card__description');
+    expect(contentElement).not.toBeNull();
+    if (contentElement) {
+      expect(contentElement.textContent).toContain(mockData.application.description);
+    }
+  });
+
+  it('should display missing content message if content is not available', () => {
+    component.application.description = '';
+    fixture.detectChanges();
+
+    const missingContentElement = fixture.nativeElement.querySelector('.app-card__description span');
+    expect(missingContentElement).not.toBeNull();
+    if (missingContentElement) {
+      expect(missingContentElement.textContent).toContain(mockData.missingContentMessage);
+    }
+  });
+
+  it('should display the correct button text', () => {
+    const buttonElement = fixture.nativeElement.querySelector('button');
+    expect(buttonElement.textContent).toContain(mockData.buttonCapture);
+  });
+
+  it('should have the correct router link value', () => {
+    const buttonElement = fixture.nativeElement.querySelector('button');
+    const routerLinkValue = buttonElement.getAttribute('ng-reflect-router-link');
+    expect(routerLinkValue).toEqual(mockData.routerLinkValue.join(','));
+  });
+
+  it('should display the picture component with the correct inputs', () => {
+    const pictureComponent = fixture.nativeElement.querySelector('app-picture');
+    expect(pictureComponent.getAttribute('ng-reflect-picture')).toEqual(mockData.application._links?.picture);
+    expect(pictureComponent.getAttribute('ng-reflect-hash-value')).toEqual(`${mockData.application.name}`);
+  });
+
+  it('should show card list', async () => {
+    const cardHarnesses = await harnessLoader.getAllHarnesses(MatCardHarness);
+    expect(cardHarnesses.length).toBe(1);
+  });
+
+  it('should display card component correctly after scrolled event', async () => {
+    const initialCardHarnesses = await harnessLoader.getAllHarnesses(MatCardHarness);
+    expect(initialCardHarnesses.length).toBe(1);
+
+    document.querySelector('mat-card')?.dispatchEvent(new Event('scrolled'));
+    fixture.detectChanges();
+
+    const updatedCardHarnesses = await harnessLoader.getAllHarnesses(MatCardHarness);
+    expect(updatedCardHarnesses.length).toBe(1);
+  });
+
+  it('should display card component correctly with search query', async () => {
+    fixture.componentInstance.application.name = 'sample query';
+    fixture.detectChanges();
+
+    const cardHarnesses = await harnessLoader.getAllHarnesses(MatCardHarness);
+    expect(cardHarnesses.length).toBe(1);
+  });
```

Comment:
🗑️  -- not pertinent to application-card component

Reply:
Done